sycl/test/basic_tests/diagnostics/handler.cpp

```diff
@@ -1,5 +1,8 @@
-// RUN: %clangxx -fsycl %s -o %t.out
-// RUN: %t.out | FileCheck %s
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out | FileCheck %s
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
```

Comment:
Are these changes needed?
https://github.com/intel/llvm/pull/1543 changed only line 1. Isn't it enough?

Is it related to the issue in sycl/test/basic_tests/get_nonhost_devices.cpp? If so, should it be resolved the same way?

Reply:
My understanding is that the test code triggers the default selector implicitly when it creates the `Queue`.  Therefore the `PLACEHOLDER` substitutions are used (for `RUN` and `CHECK`) to more explicitly control what is tested.

Reply:
```suggestion
// RUN: env SYCL_BE=%sycl_be %t.out | FileCheck %s
```
Should we really run this test on 4 different devices to validate exceptions handling?

I'd like to keep existing testing approach and align on overall testing strategy separately.

Reply:
+1 The check and exception generation is done on host side (before offloading). There is no reason to run it on every device.

Reply:
I see your point. Change with next pull.